sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1865,7 +1865,7 @@ class Dataset[T] private[sql](
     val resolver = sparkSession.sessionState.analyzer.resolver
     val allColumns = queryExecution.analyzed.output
     val groupCols = colNames.map { colName =>
-      allColumns.find(col => resolver(col.name, colName)).getOrElse(
+      allColumns.find(col => resolver(col.qualifiedName, colName)).getOrElse(
```

Comment:
Oops. My bad. Thank you for review, @cloud-fan . I will add that case and update the PR to handle both.